storage/client_test.go

```diff
@@ -29,113 +29,90 @@ import (
 var emulatorClients map[string]storageClient
 
 func TestCreateBucketEmulated(t *testing.T) {
-	checkEmulatorEnvironment(t)
-
-	for transport, client := range emulatorClients {
-		project := fmt.Sprintf("%s-project", transport)
+	transportClientTest(t, func(t *testing.T, project, bucket string, client storageClient) {
 		want := &BucketAttrs{
-			Name: fmt.Sprintf("%s-bucket-%d", transport, time.Now().Nanosecond()),
+			Name: bucket,
 		}
 		got, err := client.CreateBucket(context.Background(), project, want)
 		if err != nil {
-			t.Fatalf("%s, %v", transport, err)
+			t.Fatal(err)
 		}
 		want.Location = "US"
 		if diff := cmp.Diff(got.Name, want.Name); diff != "" {
-			t.Errorf("%s: got(-),want(+):\n%s", transport, diff)
+			t.Fatalf("got(-),want(+):\n%s", diff)
 		}
 		if diff := cmp.Diff(got.Location, want.Location); diff != "" {
-			t.Errorf("%s: got(-),want(+):\n%s", transport, diff)
+			t.Fatalf("got(-),want(+):\n%s", diff)
 		}
-	}
+	})
 }
 
 func TestDeleteBucketEmulated(t *testing.T) {
-	checkEmulatorEnvironment(t)
-
-	for transport, client := range emulatorClients {
-		project := fmt.Sprintf("%s-project", transport)
+	transportClientTest(t, func(t *testing.T, project, bucket string, client storageClient) {
 		b := &BucketAttrs{
-			Name: fmt.Sprintf("%s-bucket-%d", transport, time.Now().Nanosecond()),
+			Name: bucket,
 		}
 		// Create the bucket that will be deleted.
 		_, err := client.CreateBucket(context.Background(), project, b)
 		if err != nil {
-			// Flag the error and continue so the Delete is skipped and the next
-			// transport can run its test.
-			t.Errorf("%s: %v", transport, err)
-			continue
+			t.Fatalf("error on CreateBucket %v", err)
```

Comment:
nit, here and following: I'd just say `client.CreateBucket: %v`. Or include the bucket name if you think it would be helpful for debugging. We already know it's an error :)

Reply:
Good point, done!